fairmq/shmem/Common.h

```diff
@@ -34,12 +35,21 @@ struct SharedMemoryError : std::runtime_error { using std::runtime_error::runtim
 
 using SimpleSeqFitSegment = boost::interprocess::basic_managed_shared_memory<char,
     boost::interprocess::simple_seq_fit<boost::interprocess::mutex_family, boost::interprocess::offset_ptr<void>>,
-    boost::interprocess::null_index>;
-    // boost::interprocess::iset_index>;
+    // boost::interprocess::null_index>;
+    boost::interprocess::iset_index>;
 using RBTreeBestFitSegment = boost::interprocess::basic_managed_shared_memory<char,
     boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family, boost::interprocess::offset_ptr<void>>,
-    boost::interprocess::null_index>;
-    // boost::interprocess::iset_index>;
+    // boost::interprocess::null_index>;
+    boost::interprocess::iset_index>;
+
+struct RefCount
+{
+    std::atomic<uint16_t> count;
+};
+
+using MngRefCountPool = boost::interprocess::adaptive_pool<RefCount, boost::interprocess::managed_shared_memory::segment_manager, 1024, 2>;
```

Comment:
What would be a good value for the upper bound?

Reply:
I tried this:
```cpp
constexpr size_t maxNodesPerBlock = 4096;
constexpr size_t maxFreeBlocks = 2;

constexpr size_t nodesPerBlock() {
    return std::min(boost::interprocess::mapped_region::get_page_size() / sizeof(RefCount), maxNodesPerBlock);
}

constexpr auto numNodesPerBlock = nodesPerBlock();

using RefCountPool = boost::interprocess::adaptive_pool<RefCount, boost::interprocess::managed_shared_memory::segment_manager, nodesPerBlock, maxFreeBlocks>;
```
, but it fails the constexpr check:
`call to non-‘constexpr’ function ‘static std::size_t boost::interprocess::mapped_region::get_page_size()’ `.

Anyway, unmagicked the number a lil bit for now.

Reply:
hm, from experience I know about 4kiB, 2MiB, and 1GiB as real-world page sizes used on 64bit x86 systems. I guess, 2MiB chunk would be fine still, but GiB is clearly too much ?! so, perhaps
```cpp
// size pool chunks at least as big as a memory page, but not larger than 2MiB
constexpr auto numRefCountsPerChunk = std::min(boost::interprocess::mapped_region::get_page_size(), 2 * 1024 * 1024) / sizeof(RefCount);
```
Additionally, one could make 2MiB the default upper bound, but make it a build-time configurable as well?

Just speaking out ideas loudly.

Reply:
> , but it fails the constexpr check:
> `call to non-‘constexpr’ function ‘static std::size_t boost::interprocess::mapped_region::get_page_size()’`

Didn't see this post, ya, we should detect the page size via CMake instead of looking it up at run-time.

Reply:
hmm, this will not be enough. It could be different page sizes even for different regions. Also the `mapped_region::get_page_size()` is not the right way. As a consequence shmem page sizes will not be available at compile-time to look up.

Reply:
I'm leaving it at 4096.